cmd/argo/commands/template/lint.go

```diff
@@ -57,9 +63,10 @@ func NewLintCommand() *cobra.Command {
 			if err != nil {
 				log.Fatal(err)
 			}
-			fmt.Printf("Workflow manifests validated\n")
+			fmt.Printf("WorkflowTemplate manifests validated\n")
 		},
 	}
-	command.Flags().BoolVar(&strict, "strict", true, "perform strict workflow validatation")
+	command.Flags().BoolVar(&strict, "strict", true, "perform strict workflow template validatation")
+	command.Flags().BoolVar(&resolveRemoteTemplates, "resolve-remote-templates", false, "resolve remote templates during workflow template validatation")
```

Comment:
Is there any better flag name?

Reply:
My vote would say that `--resolve-remote-templates` is good

Reply:
Thank you for your ideas.

In the current implementation of this PR, the difference between using this flag is whether it tries to solve `TemplateRef` which means remote and it raises an error if a given linting workflow reaches any `TemplateRef`. I think we should discuss the expected behavior.

Reply:
Got it. I misunderstood the issue. you are referring `@dtaniwaki is it possible to make client-side linting not require a kubeconfig in the case where templates are not used?` @jessesuen comments.

Expected Behaviors:
1. Kubeconfig should not be loaded for normal workflow (without `TemplateRef`) 
2.  Kubeconfig should be loaded for the workflow with `TemplateRef`(no flag). 

Argo Cli should automatically detect the WF with `TemplateRef` and without `TemplateRef`and execute the above behavior.  `lint` should resolve the `TemplateRef`.
ArgoCli should provide firstclass support for this instead of passing flag.